mods/persistence/controllers/subsystems/autosave.dm

```diff
@@ -22,53 +22,73 @@ SUBSYSTEM_DEF(autosave)
 	else if(saving)
 		msg = "Currently Saving..."
 	else
-		msg = "Next Autosave in [ round(((last_save + autosave_interval) - world.time) / (1 MINUTE), 0.1)] Minutes."
+		msg = "Next Autosave in [round(((last_save + autosave_interval) - world.time) / (1 MINUTE), 0.1)] Minutes."
 	..(msg)
 
 /datum/controller/subsystem/autosave/fire()
 	AnnounceSave()
-	if(last_save + autosave_interval <= world.time)
+	if((last_save + autosave_interval) <= world.time)
 		Save()
 
 /datum/controller/subsystem/autosave/proc/Save(var/check_for_restart = TRUE)
 	if(saving)
-		message_admins(SPAN_DANGER("Attempted to save while already saving!"))
-	else
-		saves += 1
-		to_world("<font size=4 color='green'>Beginning save! Server will unpause when save is complete.</font>")
-
-		var/reset_after_save = config.autosave_auto_reset > 0 && world.time >= config.autosave_auto_reset
+		message_admins(SPAN_DANGER("Attempted autosave while already making an autosave!"))
+		return
+	var/exception/last_except = null
+	var/reset_after_save      = (config.autosave_auto_reset > 0) && (world.time >= config.autosave_auto_reset)
+	saves  += 1
+	saving = TRUE
 
+	try
+		//Announce saving start!
+		to_world(SPAN_AUTOSAVE("Beginning autosave! Server will pause until complete."))
 		if(check_for_restart && reset_after_save)
-			to_world("<font size=4 color='red'>Server is resetting after this save!</font>")
+			to_world(SPAN_AUTOSAVE_WARN("Server is restarting after this autosave!"))
+		sleep(5)
```

Comment:
The server needs a bit of time to push messages to all connected users. Otherwise you get instances where nothing is shown on some of the clients because byond doesn't flush buffered messages.